### PR TITLE
Fix read only, view state bugs

### LINKED ISSUE #249: Undo does not revert UI
`undo` only reverts applicant list state, selection is not preserved.

`view 1`, then `delete 1`, then `undo`, first index is not selected.

Unlikely that this would be spotted but its a possibility.

### PR DESCRIPTION
**Pull request by Chang-CH**
Closes #249, #250

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -278,7 +278,7 @@ Format: `undo`
     * Undoable command actions include: `add`, `edit`, `delete`, `clear`, `redo`.
 * Consecutive `undo` commands are available until the applicant list returns to its **initial state at launch** of the application.
 * If the current state of the applicant list is the initial state, the `undo` command will not be available.
-
+* `view` state will not be preserved. For instance, after `view 1`, `delete 1`, `undo` will restore the deleted applicant, but the viewed applicant in the detail panel will not be reverted.
```
Comment (yunpeng1234) [inline docs/UserGuide.md:281]: I think its best to say that the UI get reverted back to default

**Review by yunpeng1234 — CHANGES_REQUESTED** [review]
LGTM except for the tiny part if u wanna change

File: docs/UserGuide.md [hunk 2]
```diff
@@ -278,7 +278,7 @@ Format: `undo`
     * Undoable command actions include: `add`, `edit`, `delete`, `clear`, `redo`.
 * Consecutive `undo` commands are available until the applicant list returns to its **initial state at launch** of the application.
 * If the current state of the applicant list is the initial state, the `undo` command will not be available.
-
+* The applicant detail panel state will not be preserved. For instance, after `view 1`and `delete 1`, `undo` will restore the deleted applicant, but applicant detail panel will not be reverted to show the deleted applicant.
```
Comment (Jeromesyl) [inline docs/UserGuide.md:281]: "For instance, after `view 1`and `delete 1`, `undo` will restore the deleted applicant, but applicant detail panel will not be reverted to show the deleted applicant."
Maybe a better wording would be:
"For instance, after `view 1`and `delete 1`, `undo` will restore the deleted applicant to the applicant list, but the applicant detail panel will not be reverted to show the details of the deleted applicant.

**Review by Jeromesyl — CHANGES_REQUESTED** [review]

**Review by yunpeng1234 — APPROVED** [review]
Lgtm

**Review by Jeromesyl — APPROVED** [review]
LGTM

**Pull request merged**